src/native/managed/native-library.props

```diff
@@ -9,28 +9,9 @@
     <ControlFlowGuard>Guard</ControlFlowGuard>
   </PropertyGroup>
 
-  <!-- set the shared library name.  this helps the native linker correctly reference this shared
-       library in dependents -->
-  <PropertyGroup>
-    <!-- in net9.0 we can do this, but only on mobile apple platforms, not OSX -->
-    <SharedLibraryInstallName>@rpath/$(MSBuildProjectName).dylib</SharedLibraryInstallName>
-  </PropertyGroup>
-  <ItemGroup Condition="'$(TargetsOSX)' == 'true'">
-      <LinkerArg Include="-Wl,-install_name,@rpath/$(MSBuildProjectName).dylib" />
-  </ItemGroup>
```

Comment:
We generally need to specify the `@rpath/`-prefixed name on macOS for libraries. Is that done somewhere else now? What is the output for `otool -l` before and after?

Reply:
> Is that done somewhere else now?

Seems to me like an avoidable duplicate:
https://github.com/dotnet/runtime/blob/7b69459abc771fda88349204654989c69ade0592/src/coreclr/nativeaot/BuildIntegration/Microsoft.NETCore.Native.Unix.targets#L217

Reply:
> Can we use it for all apple platforms? Or are there use-cases when we explicitly don't need it?

I'd say we can. We would need to check with Xamarin workloads to ensure that it doesn't break anything.

Reply:
With 9a97e5e the behavior is:

```diff
-      cmdsize 72
-         name bin/Release/net9.0/osx-arm64/native/helloworld.dylib (offset 24)
+      cmdsize 48
+         name @rpath/helloworld.dylib (offset 24)
```

* User can skip setting install_name with `SkipInstallName`
* If user has `<LinkerArg Include="-Wl,-install_name,@rpath/TheirName" />` type of thing specified in their project, TheirName takes precedence (since there can be only one install_path)

What do you think?

cDAC (the project being modified) is currently only used for testing in the repo, which is working, so I think we can live with it without waiting for next SDK update which won't be long (10.0 alpha1).

Reply:
cc @lambdageek

Reply:
LGTM